components/ModEditFacilityOrProfessionalTopbar.vue

```diff
@@ -93,7 +111,79 @@ function setSelectedId() {
     }
 }
 
+const facilityHasUnsavedChanges = () => {
+    if (!facilityRefsBeforeChange.value) return false
+
+    const facilityBeforeChange = facilityRefsBeforeChange.value
+
+    /** This needs to be converted because to access the values of this object we do not need value.
+    But to keep their reactivity in the store we keep them as Refs **/
+    const facilitySections = facilitiesStore.facilitySectionFields as unknown as {
+        [key: string]: string
+    }
+
+    const areThereUnsavedChanges
+        = facilityBeforeChange.nameEn !== facilitySections.nameEn
+        || facilityBeforeChange.nameJa !== facilitySections.nameJa
+        || facilityBeforeChange.contact.phone !== facilitySections.phone
+        || facilityBeforeChange.contact.website !== facilitySections.website
+        || facilityBeforeChange.contact.email !== facilitySections.email
+        || facilityBeforeChange.contact.address.postalCode !== facilitySections.postalCode
+        || facilityBeforeChange.contact.address.prefectureEn !== facilitySections.prefectureEn
+        || facilityBeforeChange.contact.address.cityEn !== facilitySections.cityEn
+        || facilityBeforeChange.contact.address.addressLine1En !== facilitySections.addressLine1En
+        || facilityBeforeChange.contact.address.addressLine2En !== facilitySections.addressLine2En
+        || facilityBeforeChange.contact.address.prefectureJa !== facilitySections.prefectureJa
+        || facilityBeforeChange.contact.address.cityJa !== facilitySections.cityJa
+        || facilityBeforeChange.contact.address.addressLine1Ja !== facilitySections.addressLine1Ja
+        || facilityBeforeChange.contact.address.addressLine2Ja !== facilitySections.addressLine2Ja
+        || facilityBeforeChange.contact.googleMapsUrl !== facilitySections.googlemapsURL
+        || facilityBeforeChange.mapLatitude.toString() !== facilitySections.mapLatitude
+        || facilityBeforeChange.mapLongitude.toString() !== facilitySections.mapLongitude
+        || JSON.stringify(facilityBeforeChange.healthcareProfessionalIds)
+        !== JSON.stringify(facilitySections.healthcareProfessionalIds)
+        || facilitySections.healthProfessionalsRelations.length > 0
+
+    return areThereUnsavedChanges
+}
+
+const updateFacilityOrHealthcareProfessional = async () => {
+    // This makes the on click update the facility if the screen is EditFacility
+    if (moderationScreenStore.activeScreen === ModerationScreen.EditFacility) {
+        const checkForUnsavedChanges = facilityHasUnsavedChanges()
+
+        // This prevents us from sending a requested unnecessarily if the user has not made changes
+        if (!checkForUnsavedChanges) return
+
+        const response = await facilitiesStore.updateFacility()
```

Comment:
should we add a loader here? 
Could be in a future issue.

Reply:
Yea. Could add a loader. I am not sure of the speed of it at the moment. Of course locally it runs quickly. Maybe a future issue as this one is big enough already